akka-remote/src/main/scala/akka/remote/transport/netty/TcpSupport.scala

```diff
@@ -21,9 +21,13 @@ import com.github.ghik.silencer.silent
 /**
  * INTERNAL API
  */
-private[remote] object ChannelLocalActor extends ChannelLocal[Option[HandleEventListener]] {
-  override def initialValue(channel: Channel): Option[HandleEventListener] = None
-  def notifyListener(channel: Channel, msg: HandleEvent): Unit = get(channel).foreach { _.notify(msg) }
+private[remote] object ChannelLocalActor {
+  def notifyListener(channel: Channel, msg: HandleEvent): Unit = {
+    val listener = channel.getAttachment.asInstanceOf[HandleEventListener]
```

Comment:
use attachment

Reply:
Seems like it could be a performance improvement but I don't think we want to touch this given that classic remoting is deprecated and it has always been like this without any complaints. There could be a risk that it introduces some regression that we didn't think of.

Reply:
in our schedules,which is using Akka 2.4, when with 10x thousands of remote clients, the load is high, so I was wondering port this back to Akka 2.4 latter.

Reply:
But yes, it should be better using the new transport.

Reply:
2.4 is end-of-life so that isn't an option unless you are a Lightbend customer with an extended support deal

Reply:
But how about merging this into the current release, sorry for I forgot to make a PR before, the `ChannelHandlerContext` which has an `attachment` too.